src/lsm/tree_fuzz.zig

```diff
@@ -599,96 +601,152 @@ fn EnvironmentType(comptime table_usage: TableUsage) type {
                 scan_range.direction,
             );
 
-            // Asserting the positive space:
-            // all keys found by the scan must exist in our model.
-            var tree_value_last: ?Value = null;
-            for (tree_values) |tree_value| {
+            const model_values = model.scan(
+                scan_range.min,
+                scan_range.max,
+                scan_range.direction,
+                env.scan_results_model,
+            );
```

Comment:
Great symmetry with `tree_values`!

Reply:
Yeah, that's the whole idea with extracting a dedicated `Model`, to make sure that the interface is mechanically identical.